### PR TITLE
feat: add popover for on-behalf–of

### PR DESCRIPTION
**Pull request by reidzeibel**
closes https://github.com/AtB-AS/kundevendt/issues/16033

Pretty straightforward PR, things added:

- ref to the new label.
- useFocusEffect to trigger popover when view is rendered and screen is shown.
- text for popover.

Preview below

image

image

### CODE REVIEW

**Review by marius-at-atb — APPROVED** [review]
Lgtm!

File: src/stacks-hierarchy/Root_PurchaseOverviewScreen/components/TravellerSelection.tsx [hunk 1]
```diff
@@ -187,7 +203,15 @@ export function TravellerSelection({
             </View>
 
             {/* remove new label when requested */}
-            {isOnBehalfOfEnabled && <LabelInfo label="new" />}
+            {isOnBehalfOfEnabled && (
+              <View
+                ref={onBehalfOfIndicatorRef}
+                renderToHardwareTextureAndroid={true}
```
Comment (jorelosorio) [inline src/stacks-hierarchy/Root_PurchaseOverviewScreen/components/TravellerSelection.tsx:209]: I am just curious about this, we haven't used it before.

**Review by gorandalum — APPROVED** [review]

**Comment by jorelosorio** [comment]
@reidzeibel not sure if before a tap action on a tag was done before but looks strange without more context, is not clear that it is possible to tap on it and see more information, perhaps adds a ? icon or something. Once is clicked it should be more explicit about how it works isnt?

File: src/stacks-hierarchy/Root_PurchaseOverviewScreen/components/TravellerSelection.tsx (hunk 1 above)
Comment (reidzeibel) [reply src/stacks-hierarchy/Root_PurchaseOverviewScreen/components/TravellerSelection.tsx:209]: Sometimes the popover doesn't show in Android, we need to add that line based on this source: https://www.npmjs.com/package/react-native-popover-view#show-from-a-ref-not-working

**Comment by reidzeibel** [comment]
@jorelosorio This popover view is just to notify users that there are something to look into. About your suggestion of adding a "?", do you have any example of how would it look? maybe a rough sketch or something?

**Comment by jorelosorio** [comment]
> @jorelosorio This popover view is just to notify users that there are something to look into. About your suggestion of adding a "?", do you have any example of how would it look? maybe a rough sketch or something?

I guess that @oyvindmikkelsenatb could help us to identify it, because at least is not intuitive that we can tap on that label new at least it does not work for that for other places where is used.

File: src/stacks-hierarchy/Root_PurchaseOverviewScreen/components/TravellerSelection.tsx (hunk 1 above)
Comment (jorelosorio) [reply src/stacks-hierarchy/Root_PurchaseOverviewScreen/components/TravellerSelection.tsx:209]: Interesting, thanks for the explanation.

**Comment by gorandalum** [comment]
> > @jorelosorio This popover view is just to notify users that there are something to look into. About your suggestion of adding a "?", do you have any example of how would it look? maybe a rough sketch or something?
> 
> I guess that @oyvindmikkelsenatb could help us to identify it, because at least is not intuitive that we can tap on that label new at least it does not work like that for other places where is used.

The label isn't clickable, the whole section item is. So the question is just where on the section item to point the arrow from the popover. You feel it should be pointed somewhere else @jorelosorio?

**Comment by jorelosorio** [comment]
> > > @jorelosorio This popover view is just to notify users that there are something to look into. About your suggestion of adding a "?", do you have any example of how would it look? maybe a rough sketch or something?
> > 
> > 
> > I guess that @oyvindmikkelsenatb could help us to identify it, because at least is not intuitive that we can tap on that label new at least it does not work like that for other places where is used.
> 
> The label isn't clickable, the whole section item is. So the question is just where on the section item to point the arrow from the popover. You feel it should be pointed somewhere else @jorelosorio?

OK, I understand now, I missed then the part of the useEffect that opens it. However, would be nice explains a bit more how it works. But is not related to this task.

**Pull request merged**